test/new-e2e/agent-subcommands/flare/flare_test.go

```diff
@@ -0,0 +1,109 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package flare contains helpers and e2e tests of the flare command
+package flare
+
+import (
+	_ "embed"
+	"strings"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/client/flare"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/params"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/stretchr/testify/assert"
+)
+
+type commandFlareSuite struct {
+	e2e.Suite[e2e.FakeIntakeEnv]
+}
+
+func TestFlareSuite(t *testing.T) {
+	e2e.Run(t, &commandFlareSuite{}, e2e.FakeIntakeStackDef(nil), params.WithDevMode())
+}
+
+func waitForAgentAndGetFlare(v *commandFlareSuite, flareArgs ...client.AgentArgsOption) flare.Flare {
+	_ = v.Env().Agent.Flare(flareArgs...)
+
+	flare, err := v.Env().Fakeintake.Client.GetLatestFlare()
+	assert.NoError(v.T(), err)
+
+	return flare
+}
+
+func (v *commandFlareSuite) TestFlareDefaultFiles() {
+	v.UpdateEnv(e2e.FakeIntakeStackDef(nil))
+	flare := waitForAgentAndGetFlare(v, client.WithArgs([]string{"--email", "e2e@test.com", "--send"}))
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
For readability, I would rather act and then collect data, in other words I would call

```golang

// act
v.Env().Agent.Flare(..)

// assert
flare, err := v.Env().Fakeintake.Client.GetLatestFlareEventually()
...
```

Reply:
I don't get your point sorry, do you want me to not call a function and directly call `Flare()` and `GetLatestFlare()` in each test scenario ?